testdata/go/v3/memcached-operator/monitoring/metricsdocs/metricsdocs.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2022.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"github.com/example/memcached-operator/monitoring"
+	"sort"
+)
+
+// constant parts of the file
+const (
+	title      = "# Memcached Operator Metrics\n"
+	background = "This document aims to help users that are not familiar with metrics exposed by the memcached-operator.\n" +
+		"All metrics documented here are auto-generated by the utility tool `monitoring/metricsdocs` and reflects exactly what is being exposed.\n\n"
+
+	KVSpecificMetrics = "## Memcached Operator Metrics List\n"
+
+	opening = title +
+		background +
+		KVSpecificMetrics
+
+	// footer
+	footerHeading = "## Developing new metrics\n"
+	footerContent = "After developing new metrics or changing old ones, please run `make generate-metricsdocs` to regenerate this document.\n\n" +
+		"If you feel that the new metric doesn't follow these rules, please change `monitoring/metricsdocs` according to your needs.\n"
+
+	footer = footerHeading + footerContent
+)
+
+// MemcachedMetricList contains the name, description, and type for each metric.
+func main() {
```

Comment:
since [metricsdocs.go](https://github.com/operator-framework/operator-sdk/pull/6114/files#diff-29567a3dab531ff74b6e68ff734786bb53df8bee629bc1d8eaa33809db03105f) is not actually part of the operator code, I think it would make more sense and look cleaner if we moved the script to generate the documentation under a new `tools` folder, maybe `tools/monitoring/metricsdocs.go`

Reply:
@machadovilaca Since we recommend to have the monitoring code in a separate sub folder, I would suggest 
`monitoring/tools/metricsdocs.go`. What do you think?

Reply:
okay, I think it already makes a clearer distinction

Reply:
Done. now it's: 
**monitoring/tools/metricsdocs.go**
The doc file:
docs/monitoring/metrics.md
the metrics file:
monitoring/metrics.go